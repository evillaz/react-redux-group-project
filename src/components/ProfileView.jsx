import { useSelector } from 'react-redux';

const ProfileView = () => {
  const RocketData = useSelector((state) => state.rocket.RocketDatile.filter((item) => {
    if (item.reserved) {
      return true;
    }
    return false;
  }));

  const myMissions = useSelector((store) => store.missions.missions.filter((mission) => {
    if (mission.reserved) {
      return true;
    }
    return false;
  }));

  return (
    <div className="profile-container">
      <div className="my-subtab">
        <h3 className="profile-heading">My Missions</h3>
        <ul className="feature-list">
          {myMissions.length > 0
            ? (
              myMissions.map((mission) => (
                <li className="feature-item" key={mission.mission_id}>{mission.mission_name}</li>
              ))
            )
            : (
              <div className="feature-item">
                You haven&apos;t joined any Missions yet
              </div>
            )}
        </ul>
      </div>
      <div className="my-subtab">
        <h3 className="profile-heading">My Rockets</h3>
        <ul className="feature-list">
          {RocketData.length > 0 ? (
            RocketData.map((item) => (
              <li className="feature-item" key={item.id}>{item.name}</li>
            ))
          ) : (
            <li className="feature-item" style={{ border: 'none' }}>not rocket yet</li>
          )}

        </ul>

      </div>
    </div>

  );
};

export default ProfileView;
