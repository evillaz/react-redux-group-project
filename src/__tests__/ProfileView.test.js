import React from 'react';
import { useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import ProfileView from '../components/ProfileView';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('ProfileView.jsx', () => {
  it('Testing of ProfileView.jsx', () => {
    useSelector.mockImplementation((selectorFn) => selectorFn({
      missions: {
        missions: [
          { mission_id: 1, mission_name: 'Mission 1', reserved: true },
          { mission_id: 2, mission_name: 'Mission 2', reserved: false },
        ],
      },
      rocket: {
        RocketDatile: [
          { id: 1, name: 'Rocket 1', reserved: true },
          { id: 2, name: 'Rocket 2', reserved: false },
        ],
      },
    }));
    const { container } = render(<ProfileView />);
    expect(container).toMatchSnapshot();
  });
});
