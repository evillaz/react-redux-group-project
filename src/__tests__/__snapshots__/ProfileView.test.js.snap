// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['ProfileView.jsx Testing of ProfileView.jsx 1'] = `
<div>
  <div
    class="profile-container"
  >
    <div
      class="my-subtab"
    >
      <h3
        class="profile-heading"
      >
        My Missions
      </h3>
      <ul
        class="feature-list"
      >
        <li
          class="feature-item"
        >
          Mission 1
        </li>
      </ul>
    </div>
    <div
      class="my-subtab"
    >
      <h3
        class="profile-heading"
      >
        My Rockets
      </h3>
      <ul
        class="feature-list"
      >
        <li
          class="feature-item"
        >
          Rocket 1
        </li>
      </ul>
    </div>
  </div>
</div>
`;
